src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.report;
+
+import core.basesyntax.operation.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private ShopService shopService;
+
+    @Override
+    public String generateReport() {
+        StringBuilder builder = new StringBuilder("Fruit, Quantity\n");
+        for (Map.Entry<String, Integer> entry: shopService.getFruits().entrySet()) {
+            builder.append(entry.getKey())
+                    .append(",")
+                    .append(entry.getValue())
+                    .append("\n");

```

Comment:
Avoid hardcoding in your solution. You have hardcoded the report structure here. It's better to make it more flexible, for example, by passing the structure as a parameter or by using a more flexible data structure.